src/flask/typing.py

```diff
@@ -33,14 +33,20 @@
     "WSGIApplication",
 ]
 
+GenericException = t.TypeVar("GenericException", bound=Exception, contravariant=True)
+
 AppOrBlueprintKey = t.Optional[str]  # The App key is None, whereas blueprints are named
 AfterRequestCallable = t.Callable[["Response"], "Response"]
 BeforeRequestCallable = t.Callable[[], None]
-ErrorHandlerCallable = t.Callable[[Exception], ResponseReturnValue]
 TeardownCallable = t.Callable[[t.Optional[BaseException]], "Response"]
 TemplateContextProcessorCallable = t.Callable[[], t.Dict[str, t.Any]]
 TemplateFilterCallable = t.Callable[[t.Any], str]
 TemplateGlobalCallable = t.Callable[[], t.Any]
 TemplateTestCallable = t.Callable[[t.Any], bool]
 URLDefaultCallable = t.Callable[[str, dict], None]
 URLValuePreprocessorCallable = t.Callable[[t.Optional[str], t.Optional[dict]], None]
+
+
+class ErrorHandlerCallable(t.Protocol[GenericException]):
+    def __call__(self, error: GenericException, /) -> ResponseReturnValue:
```

Comment:
Positional-only arguments (with the use of `/`) has been introduced in Python 3.8 so this code is not compatible with earlier interpreters. However it gives the benefit of not enforcing the name of the argument `error`.

If that's preferable I can also introduce some code that depends on the current version of `Python` and give another line before `3.8`

Reply:
Apparently readthedocs doesn't accept this ...

Reply:
I don't see how RTD is related here.

Anyway, I think introducing syntax that doesn't work on all supported Python versions is not a good idea (I guess type checking would fail on those versions?)

Reply:
We need to wait until we drop Python 3.7 in 2023. Typing supports the same versions the code supports.

Reply:
Yes, Read The Docs is parsing the python using python 3.7 thus choking on the `/`. Let me try to use a python version check before using the syntax and fallback on imposing the `error` name for the argument for earlier versions.

Reply:
fwiw we could use 3.9 for the RTD build. but if we need a version check anyway there's probably no need for that

Reply:
It look like Python still wants to be able to parse the protected code anyway, and I don't blame it: it could hide some open parenthesis. I'll drop my idea of letting the users choose their argument name: they'll have to stick to `error`